code/controllers/subsystems/cleanup_ch.dm

```diff
@@ -0,0 +1,165 @@
+#define FIRE_PRIORITY_CLEANUP 20
+#define MAX_CLEANUP_QUEUE_TIME 120 //Max time in queue before we give up.
+#define URGENCY_PANIC_MODE 0.75 //Max urgency before we start to skip checks
+#define SMOOTHSTEP(variable) (max(min(6*(variable**5) - 15*(variable**4) + 10*(variable**3),1),0))
+//This is the cleanup subsystem. Its responsibility is to clean up unneeded atoms such as empty bullet casings and dead mobs in an effort to boost performance.
+SUBSYSTEM_DEF(cleanup)
+	name = "Cleanup"
+	priority = FIRE_PRIORITY_CLEANUP
+	flags = SS_BACKGROUND|SS_NO_INIT
+	wait = 1200 //Fires every 2 minutes.
+
+	var/stat_tag = "Cl" //Used for logging
+	var/urgency //How desperate are we to delete stuff?
+	var/points //How much complexity is in our queue?
+	var/list/processing = list()
+	var/list/currentrun = list()
+
+	var/datum/cleanupentry/current_thing
```

Comment:
Just curious, is this only used for the Recover()?

Reply:
If we're talking about current_thing, then it's used for Recover and in the actual fire proc, I believe all of these were just copied from other subsystems though to sort of follow the standard for them.